huxley/core/models.py

```diff
@@ -361,6 +361,20 @@ def remove(assignment_data):
 
         return failed_assignments
 
+    @classmethod
+    def update_assignment(cls, **kwargs):
+        '''Ensures that when an assignment's school field changes,
+           any delegates assigned to that assignment are no longer
+           assigned to it and that its rejected field is false.'''
+        assignment = kwargs['instance']
+        if not assignment.id:
+            return
+
+        old_assignment = cls.objects.get(id=assignment.id)
+        if not assignment.school.id == old_assignment.school.id:
```

Comment:
two things:
- `foo != bar` is more readable than `not foo == bar`
- Accessing the `school` property of the assignment triggers a DB fetch of the school object. Instead, just use the `school_id` property, since that's all we need: https://docs.djangoproject.com/en/1.8/topics/db/optimization/#use-foreign-key-values-directly